ts/output/common/FontData.ts

```diff
@@ -225,6 +233,51 @@ export type FontParameters = {
   extra_ic: number
 };
 
+/****************************************************************************/
+
+/**
+ * A function for setting up an additional character-data file
+ */
+export type DynamicSetup = ((font: FontData<CharOptions, VariantData<CharOptions>, DelimiterData>) => void);
+
+/**
+ * Character numbers or ranges of numbers that cause a dynamic file to be laoded
+ */
+export type DynamicRanges = (number | [number, number])[];
+
+/**
+ * List of characters (number) or ranges ([number, number]) of characters for
+ *   each variant in a dynamic font file
+ */
+export type DynamicVariants = {[name: string]: DynamicRanges};
+
+/**
+ * Name and variant data for a dynamic font file
+ */
+export type DynamicFileDef = [string, DynamicVariants, DynamicRanges?];
+
+/**
+ * Data stored about a dynamic font
+ */
+export type DynamicFile = {
+  file: string;                   // file containing the character data
+  variants: DynamicVariants;      // characters in each variant in the file
+  delimiters: DynamicRanges;      // delimiters in the file
+  setup: DynamicSetup;            // setup function to call to add the characters to the font
+  promise: Promise<void>;         // promise for when the file is loaded
+  failed: boolean;                // true when loading has failed
+};
+
+/**
+ * Object listing dynamic file data indexed by file name
+ */
+export type DynamicFileList = {[name: string]: DynamicFile};
+
+/**
+ * Map of characters that load a dynamic file
+ */
+export type DynamicCharMap = {[name: number]: DynamicFile};
```

Comment:
This should really be a JavaScript Map.

Reply:
The reason this and the other map that you indicate (and a number of other ones like `CharMap` and `VariantMap`) are objects rather than `Map` instances is that these are used in object literals when the fonts are bing defined.  E.g., the `delimiter.ts` file for mathjax-modern begins

``` javascript
import {DelimiterMap, DelimiterData} from 'mathjax-full/js/output/chtml/FontData.js';
import {V, H} from 'mathjax-full/js/output/common/Direction.js';

export const delimiters: DelimiterMap<DelimiterData> = {
  0x28: {
    dir: V,
    sizes: [.997, 1.095, 1.195, 1.445, 1.793, 2.093, 2.393, 2.991],
    stretch: [0x239B, 0x239C, 0x239D],
    HDW: [.748, .248, .875]
  },
  0x29: {
    dir: V,
    sizes: [.997, 1.095, 1.195, 1.445, 1.793, 2.093, 2.393, 2.991],
    stretch: [0x239E, 0x239F, 0x23A0],
    HDW: [.748, .248, .875]
  },
  0x2D: {
    c: 0x2212,
    dir: H,
    stretch: [0, 0x2212],
    HDW: [0.583, 0.083, .778]
  },
  ...
```

which is a `DelimiterMap`.  While it would be possible to use a `Map` constructor here, via something like

``` javascript
export const delimiters: DelimiterMap<DelimiterData> = new Map([
  [0x28, {
    dir: V,
    sizes: [.997, 1.095, 1.195, 1.445, 1.793, 2.093, 2.393, 2.991],
    stretch: [0x239B, 0x239C, 0x239D],
    HDW: [.748, .248, .875]
  }],
  [0x29, {
    dir: V,
    sizes: [.997, 1.095, 1.195, 1.445, 1.793, 2.093, 2.393, 2.991],
    stretch: [0x239E, 0x239F, 0x23A0],
    HDW: [.748, .248, .875]
  }],
  [0x2D, {
    c: 0x2212,
    dir: H,
    stretch: [0, 0x2212],
    HDW: [0.583, 0.083, .778]
  }],
  ...
]);
```

this adds extra characters (two for each delimiter), making the file longer, and to see is less clear.  Also, there was a feature request some time ago about turning the font data into stringified JSON data, as apparently it is faster to JSON parse the string than to have the javascript compiler have to compile the large object.  I guess that could still be done for the array being passed to `new Map()`.  But this means I need to rewrite the font-generation code and rebuild all the font packages.  I'd rather put that off until later in case there are other changes that need to be made to the fonts.

Also, I mentioned `CharMap` and `VariantMap`, and there are a number of other such objects used in the font definitions.  It would be good to handle those all at once, so I'd like to put this off for now.  I'm also not sure if any of the PRs that are built on top of this one will need to be modified if we make the changes here, so I thin kit would be best to get all the font PRs merged, and then do the change over to maps, if that seems to be the way to go.

Is that acceptable?

Reply:
That does make sense. 
Ultimately, it might be a good idea to hide the maps in a wrapping class, with a corresponding `fromJson` and `toJson` method.